### PR TITLE
RQM Field Mapping Optimized

### PR DESCRIPTION
**Pull request by prashelke**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [x] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced data retrieval capabilities for online item fields, improving efficiency and user experience.
	- Introduced a new `TypeIdentifier` property to better represent field information within the application.

- **Bug Fixes**
	- Improved error handling for data fetching processes, ensuring better logging and performance.

### CODE REVIEW

File: Ginger/GingerCoreCommon/Repository/ALMLib/ExternalItemFieldBase.cs [hunk 1]
```diff
@@ -40,6 +40,8 @@ public static class Fields
         [IsSerializedForLocalRepository]
         public string ID { get; set; }
 
+        public string TypeIndetifier { get; set; }
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/Repository/ALMLib/ExternalItemFieldBase.cs:43]: Please fix the spelling of this property

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs [hunk 2]
```diff
@@ -1151,161 +1153,228 @@ public static ObservableList<ExternalItemFieldBase> GetOnlineItemFields(Backgrou
...
                         }
                     }
+
+
+
+                    #endregion
+
+                    #region Get Values for category type Prvious code for referance
+
+                    //RqmResponseData category = RQM.RQMConnect.Instance.RQMRep.GetRqmResponse(loginData, new Uri(rqmSserverUrl + RQMCore.ALMProjectGroupName + "/service/com.ibm.rqm.integration.service.IIntegrationService/resources/" + ALMCore.DefaultAlmConfig.ALMProjectGUID + "/category"));
```
Comment (Maheshkale447) [inline Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:1227]: Remove unwanted code

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs [hunk 3]
```diff
@@ -1151,161 +1153,228 @@ public static ObservableList<ExternalItemFieldBase> GetOnlineItemFields(Backgrou
                                 itemfield.ToUpdate = false;
                                 itemfield.Mandatory = false;
                             }
-
+                            Reporter.ToLog(eLogLevel.DEBUG, $"field name:{itemfield.Name} field Id ={itemfield.ID} field Type ={itemfield.Type} field mandetory ={itemfield.Mandatory} field ItemType={itemfield.ItemType} field toupdate= {itemfield.ToUpdate}");
```
Comment (Maheshkale447) [inline Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:1156]: Remove the log if its not needed

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs [hunk 4]
```diff
@@ -924,43 +925,44 @@ private static void SaveItemFields(ObservableList<ExternalItemFieldBase> refresh
...
-            int maxPageNumber_ = 0;
-            int totalCategoryTypeCount = 0;
+                //------------------------------- Improved solution
 
+                string baseUri_ = string.Empty;
+                string selfLink_ = string.Empty;
+                int maxPageNumber_ = 0;
+                int totalCategoryTypeCount = 0;
+
+
```
Comment (Maheshkale447) [inline Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:948]: Remove unwanted code

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs [hunk 5]
```diff
@@ -924,43 +925,44 @@ private static void SaveItemFields(ObservableList<ExternalItemFieldBase> refresh
...
 
-            string categoryValue = string.Empty;  // --> itemfield.PossibleValues.Add(ccNode.Name);
-            string categoryTypeID = string.Empty; //--> itemfield.ID
-            try
-            {
                 //TODO: Populate list fields with CategoryTypes
+                Reporter.ToLog(eLogLevel.DEBUG, $"Starting fields retrieve process... ");
                 populatedValue = "Starting fields retrieve process... ";
-                if(bw != null)
+                if (bw != null)
```
Comment (Maheshkale447) [inline Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:955]: use the function created for populating this message on UI and use it everywhere

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs [hunk 6]
```diff
@@ -1004,7 +1006,7 @@ public static ObservableList<ExternalItemFieldBase> GetOnlineItemFields(Backgrou
                     }
 
                     //Parallel computing solution
-                    List<XmlNode> entryList = new List<XmlNode>();
+                    List<XmlNode> entryList = new();
                     if (categoryTypeUriPages.Count > 1)
                     {
```
Comment (Maheshkale447) [inline Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:1011]: Please check if code can be refactored as most of the code block in if is duplicated in else block as well

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs (hunk 3 above)
Comment (prashelke) [reply Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:1156]: done

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs (hunk 5 above)
Comment (prashelke) [reply Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:955]: done

File: Ginger/GingerCoreCommon/Repository/ALMLib/ExternalItemFieldBase.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreCommon/Repository/ALMLib/ExternalItemFieldBase.cs:43]: done

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs (hunk 4 above)
Comment (prashelke) [reply Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:948]: done

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs (hunk 2 above)
Comment (prashelke) [reply Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:1227]: done

File: Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs [hunk 7]
```diff
@@ -1151,161 +1153,228 @@
...
                         }
                     }
+
+
+
+                    #endregion
+
+                    #region Get Values for category type Prvious code for referance
+
+                    //RqmResponseData category = RQM.RQMConnect.Instance.RQMRep.GetRqmResponse(loginData, new Uri(rqmSserverUrl + RQMCore.ALMProjectGroupName + "/service/com.ibm.rqm.integration.service.IIntegrationService/resources/" + ALMCore.DefaultAlmConfig.ALMProjectGUID + "/category"));
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCore/ALM/RQM/ImportFromRQM.cs:1227]: done

**Pull request merged**